src/schema.js

```diff
@@ -179,6 +180,35 @@ class Schema {
     });
   }
 
+  mixinNamedParamFacade(api, def) {
+    function namedParamFacade(base, paramDef, ...args) {
+      let params = args;
+      if (args && args.length === 1 && typeof args[0] === 'object') {
+        const defaults = paramDef.reduce((result, item) => {
+          result[item.Name] = item.DefaultValue;
+          return result;
+        }, {});
+
+        const isNamedParameterObject = Object.keys(args[0]).reduce((result, key) =>
+          result && Object.prototype.hasOwnProperty.call(defaults, key), true);
+
+        if (isNamedParameterObject) {
+          Object.keys(args[0]).forEach((key) => { defaults[key] = args[0][key]; });
+          params = Object.keys(defaults).map(key => defaults[key]);
+        }
+      }
+      return base.apply(this, params);
+    }
+
+    Object.keys(def).forEach((key) => {
+      const fnName = key.substring(0, 1).toLowerCase() + key.substring(1);
```

Comment:
Could we move this out to a `camelCasing` function that is used in `generateDefaultApi` as well?